pkg/controller/pravega/pravega_controller.go

```diff
@@ -106,6 +117,37 @@ func makeControllerPodSpec(p *api.PravegaCluster) *corev1.PodSpec {
 			},
 		}
 		volumes = append(volumes, v)
+
+		m := corev1.VolumeMount{
+			Name:      heapDumpName,
+			MountPath: heapDumpDir,
+		}
+		volumeMounts = append(volumeMounts, m)
+	}
+	if _, ok = p.Spec.Pravega.Options["configMapVolumeMounts"]; ok {
+		configMapVolumeMounts = strings.Split(p.Spec.Pravega.Options["configMapVolumeMounts"], ",")
+		for _, vm := range configMapVolumeMounts {
+			p := strings.Split(vm, "=")
+			s := strings.Split(p[0], ":")
+			v := corev1.Volume{
+				Name: s[0],
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: s[0],
+						},
+					},
+				},
+			}
+			volumes = append(volumes, v)
+
+			m := corev1.VolumeMount{
+				Name:      s[0],
+				MountPath: p[1],
+				SubPath:   s[1],
+			}
```

Comment:
could u please explain why is the subpath being set here?
could we not instead set the volume mount in the following manner
```
Name:      s[1],
MountPath: p[1],
```

Reply:
Hi Srishti, if we don't use subPath and only use the name and mountPath for the configMap volume mount here, the ConfigMap value, i.e. content of our version of logback conf, will not override the file specified in `MountPath`, i.e. `/opt/pravega/conf/logback.xml`, as expected. Instead, it will override the file path `MountPath\<configKey>`, i.e. `/opt/pravega/conf/logback.xml/logback.xml`, which is not what we expected.

On the other hand, there are many existing pravega confs in `/opt/pravega/conf/` and we only want to override the logback conf `/opt/pravega/conf/logback.xml`.

To meet our requirment, the use of subpath here is our only choice.